### PR TITLE
Test many format/type combos for ReadPixels.

### PR DESCRIPTION
**Pull request by jdashg**

### CODE REVIEW

**Comment by jdashg** [comment]
Blink currently allows ReadPixels(RGBA/UNSIGNED_SHORT) from the RGBA8-ish backbuffer. @kenrussell

**Comment by jdashg** [comment]
This would be allowed, I suppose, if the backbuffer were RGBA16, but I think that would break a lot of other interactions?

File: sdk/tests/conformance/reading/read-pixels-test.html [hunk 1]
```diff
@@ -206,37 +199,113 @@
...
         }
-      ]);
+      );
+    }
+
+    const ext = gl.getExtension('OES_texture_half_float');
+    if (ext) {
+      TYPES.push(
+        {
+          type: gl.HALF_FLOAT_OES,
```
Comment (kenrussell) [inline sdk/tests/conformance/reading/read-pixels-test.html:286]: This should be `ext.HALF_FLOAT_OES`. Please test the WebGL 1.0 version of this modification. e.g.:

http://localhost:8080/sdk/tests/conformance/reading/read-pixels-test.html

**Review by kenrussell — APPROVED** [review]
LGTM with one bug fix.

**Pull request merged**